src/nrniv/nrnbbcore_write.cpp

```diff
@@ -344,6 +353,30 @@ static void part2_clean() {
     delete artdata2index_;
     artdata2index_ = NULL;
   }
+
+  // clean up the art Memb_list of CellGroup[].mlwithart
+  // But if multithread and direct transfer mode defer deletion of
+  // data for artificial cells.
+  if (corenrn_direct && nrn_nthread > 0) {
+    deferred_type2artdata.resize(nrn_nthread);
+  }
+  for (int ith=0; ith < nrn_nthread; ++ith) {
+    MlWithArt& mla = cellgroups_[ith].mlwithart;
+    for (size_t i = 0; i < mla.size(); ++i) {
+      int type = mla[i].first;
+      Memb_list* ml = mla[i].second;
+      if (nrn_is_artificial_[type]) {
+        if (!deferred_type2artdata.empty()) {
+          deferred_type2artdata[ith][type] = Deferred_pair(ml->nodecount, ml->data);
```

Comment:
So I was able to get rid of the ```Deferred_map```. But do I really need a ```Deferred_pair``` for this use. I know I could have eliminated it with an explicit  ```std::pair(int, double**)``` but is there even a better syntax here given that the left side knows what it expects?

Reply:
alternative could be [initialiser list](https://stackoverflow.com/questions/3250123/what-would-a-stdmap-extended-initializer-list-look-like):

```
deferred_type2artdata[ith][type] = {ml->nodecount, ml->data};
```

Reply:
Thanks!